File: backend/models/host.go
package models

import "time"

const TENANT_MODEL_NAME = "tenants"
const TENANT_CREATOR_MODEL_NAME = "tenant_creators"

type Tenant struct {
	ID                       uint      `gorm:"primary_key;column:id" json:"id"`
	Name                     string    `gorm:"column:name;not null" json:"name" validate:"required"`
	TenantUrl                string    `gorm:"column:tenantUrl;not null" json:"tenantUrl" validate:"required"`
	CreatedAt                time.Time `gorm:"column:createdAt; default:current_timestamp" json:"createdAt"`
	TenantDBConnectionString string    `gorm:"column:tenantDBConnectionString;not null;unique" json:"tenantDBConnectionString" validate:"required"`
	UserID                   uint      `json:"userId" gorm:"column:userId;not null" validate:"required"`
	User                     *User     `json:"user" gorm:"column:userId;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;" validate:""`
}

func (*Tenant) TableName() string {
	return TENANT_MODEL_NAME
}

File: frontend/shared/types/host.ts
import { SqlID, Time } from './base';

export type Tenant = {
  id: SqlID;
  name: string;
  tenantUrl: string;
  createdAt: Time;
  tenantDBConnectionString: string;
  tenantOwnerId: SqlID;
  tenantOwner?: TenantOwner;
};

export type TenantOwner = {
  id: SqlID;
  name: string;
  email: string;
  password: string;
  createdAt: Time;
};
